### PR TITLE
feat: adds flush api

### PR DESCRIPTION
**Pull request by eaddingtonwhite**

### CODE REVIEW

**Review by cprice404 — COMMENTED** [review]
Thanks for doing this!  @malandis has a better eye for the python code than I do so I will defer to him on final approval but I looked over it and didn't spot any issues!

**Review by malandis — COMMENTED** [review]
Looks great. One question about documentation.

**Review by malandis — COMMENTED** [review]
Looks great. One question about documentation.

File: src/momento/cache_client_async.py [hunk 1]
```diff
@@ -214,15 +215,26 @@ async def delete_cache(self, cache_name: str) -> DeleteCacheResponse:
...
 
+    async def flush_cache(self, cache_name: str) -> CacheFlushResponse:
+        """Flushes a cache and empties it of all its items.
+
+        Args:
+            cache_name (str): Name of the cache to be flushed.
+
+        Returns:
+            CacheFlushResponse: The result of a Flush Cache operation.
+        """
```
[the remark this answers is not in the document]
Comment (eaddingtonwhite) [reply src/momento/cache_client_async.py:230]: I can update wording. We should probably audit other SDK's as well then if we want to call this out explicitly.
https://github.com/momentohq/client-sdk-dotnet/blob/main/src/Momento.Sdk/ICacheClient.cs#L63-L83C1

[the remark this answers is not in the document]
Comment (cprice404) [reply src/momento/cache_client_async.py:230]: i would say let's just keep this one consistent with the docs we have in the other SDks for now, until/unless it surfaces as an issue for someone.  The DDB writes are transactional and I think there may even be a delay in the server side code before it returns so I think it's effectively synchronous and don't want to add undue cognitive burden for the users.

[the remark this answers is not in the document]
Comment (eaddingtonwhite) [reply src/momento/cache_client_async.py:230]: Ok sounds good will leave then. @malandis any issues with what we have now beyond async call out then?

**Review by malandis — APPROVED** [review]

**Pull request merged**